### DISCUSSION TITLE
FATAL: kernel too old

### DISCUSSION THREAD

**Discussion by chawyehsu** [Homebrew/discussions#3802] [Linux] [answered]
I had Linuxbrew installed before the [Linuxbrew-core migration](https://github.com/Homebrew/linuxbrew-core). It ran perfectly without any issues. Today after the migration via `brew update` and `brew upgrade`, all packages give the same **FATAL: kernel too old** error.

I could not find a solution to fix it, *can't find any related issues with the keyword 'FATAL: kernel too old'*, then I uninstalled and did a fresh installation of brew, by:

```sh
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```

The installation worked fine, and brew and it's commands run as expected. But when I install and use arbitrary packages, all of them give the **FATAL: kernel too old**. Below is my machine's specification and I believe it satisfies [brew's requirement](https://docs.brew.sh/Homebrew-on-Linux#requirements).

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.7 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.7 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
$ uname -a
Linux Server 2.6.32-042stab142.1 #1 SMP Tue Jan 28 23:44:17 MSK 2020 x86_64 x86_64 x86_64 GNU/Linux
```

brew commands

```
Server ~
$ brew doctor
Your system is ready to brew.
Server ~
$ which xz
/home/linuxbrew/.linuxbrew/bin/xz
Server ~
$ xz
FATAL: kernel too old
Server ~
$ which zstd
/home/linuxbrew/.linuxbrew/bin/zstd
Server ~
$ zstd
FATAL: kernel too old
Server ~
$ which gcc-12
/home/linuxbrew/.linuxbrew/bin/gcc-12
Server ~
$ gcc-12
FATAL: kernel too old
Server ~
$ brew config
HOMEBREW_VERSION: 3.6.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: fb52a12144e80118c112ea744ab65ab6ca997d67
Last commit: 8 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 499234bb176ecb6968c3df29e79d2431975e120c
Core tap last commit: 18 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nano
HOMEBREW_MAKE_JOBS: 2
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: dual-core 64-bit sandybridge
Clang: N/A
Git: 2.7.4 => /usr/bin/git
Curl: 7.47.0 => /usr/bin/curl
Kernel: Linux 2.6.32-042stab142.1 x86_64 GNU/Linux
OS: Ubuntu 16.04.7 LTS (xenial)
Host glibc: 2.23
/usr/bin/gcc: 5.4.0
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

Any way to fix it?

**Comment by carlocab** [comment]
CC @Homebrew/linux

**Comment by Bo98** [comment, marked as answer]
I believe newer Glibc require Linux kernel 3.2, so the documentation is likely outdated. Ubuntu 16.04 should install Linux kernel 4.4 though by default.

It looks like your Linux kernel may be linked to an OpenVZ 6 setup you have, which was EOL 3 years ago. Anything OpenVZ 7 or later should work (latest is OpenVZ 9).

**Comment by chawyehsu** [comment]
Thanks, I'll remove brew on the machine according to this information. It's true the kernel is linked to OpenVZ 6 and there is no way for me to upgrade the kernel to a newer version.

Besides, perhaps you need to update the documentation to present the latest requirements. [What's more, some outdated code in the formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/glibc.rb#L25).
